integration/utils/follower.go

```diff
@@ -55,13 +54,10 @@ type txFollowerImpl struct {
 
 	interval time.Duration
 
-	inprogress *atomic.Int64
-
-	mu      *sync.RWMutex
-	height  uint64
-	blockID flowsdk.Identifier
-
-	txToChan sync.Map
+	mu       *sync.RWMutex
+	height   uint64
```

Comment:
nit: could you add `// guarded by mu` to each applicable fields in case the fields got moved around / newline added / etc.